### PR TITLE
Refactor RecurlyUrlManager to return Url object instead of string.

### PR DESCRIPTION
**Pull request by tomflanagan**
Issue: https://www.drupal.org/node/2642120

Refactored RecurlyUrlManager::hostedUrl to return Drupal/Core/Url Object. I then found all instances in the code that called that method and chained Drupal/Core/Url::getUri.

### CODE REVIEW

**Comment by adamzimmermann** [comment]
So what does this change do for us that is better than how it was before? Just want to be learning the best practice ways of OOO land.

**Comment by tomflanagan** [comment]
@adamzimmermann I believe the reasoning for this refactor is by returning the object, we can access it's other properties and methods if we need to. Although right now we want to just access the url, that could change in new areas of the code.

@markdorison Correct me if that wasn't the reasoning.

**Comment by adamzimmermann** [comment]
`->getUri()` returns a string though, so aren't you limiting what you are returning to a string?

**Comment by tomflanagan** [comment]
Well in that instance we need the string. `->getUri()` is a method of `Drupal/Core/Url` which is the object we return. Now let's say I want to get access '->access(AccountInterface $account = NULL)`, I can still call the same RecurlyUrlManager::hostedUrl function to give me the url object vs have to write another method that does basically the same thing except returns the result from another method of the url class. Does that make sense?

File: src/RecurlyUrlManager.php [hunk 1]
```diff
@@ -25,6 +25,6 @@ public function hostedUrl($path = '', $subdomain = NULL) {
       $subdomain = \Drupal::config('recurly.settings')->get('recurly_subdomain');
     }
 
-    return \Drupal\Core\Url::fromUri('https://' . $subdomain . '.recurly.com/' . $path)->getUri();
+    return \Drupal\Core\Url::fromUri('https://' . $subdomain . '.recurly.com/' . $path);
```
Comment (adamzimmermann) [inline src/RecurlyUrlManager.php:28]: Ah so this is where you are just returning the object and this must help in other locations, but then you have to compensate by rendering this into a string in all of those other places?

Comment (tomflanagan) [inline src/RecurlyUrlManager.php:28]: Correct.

Comment (markdorison) [inline src/RecurlyUrlManager.php:28]: The method docs need to be updated here.

Comment (tomflanagan) [inline src/RecurlyUrlManager.php:28]: Method docs have been updated.

**Comment by markdorison** [comment]
Looks good for a D.O. patch. :+1:

**Comment by markdorison** [comment]
Patch posted/committed. https://www.drupal.org/node/2642120

**Pull request closed without merging**